enterprise/server/remote_execution/runner/runner.go

```diff
@@ -374,7 +375,37 @@ func (r *commandRunner) Run(ctx context.Context) *interfaces.CommandResult {
 		return r.sendPersistentWorkRequest(ctx, command)
 	}
 
-	return r.Container.Exec(ctx, command, &commandutil.Stdio{})
+	execResult := r.Container.Exec(ctx, command, &commandutil.Stdio{})
+
+	// If the runner has exceeded 90% of memory or disk usage, don't try to recycle
+	// it, because that may cause failures if it's reused
+	maxedOutStr := ""
+	for _, fsUsage := range execResult.UsageStats.GetPeakFileSystemUsage() {
+		if float64(fsUsage.UsedBytes)/float64(fsUsage.TotalBytes) >= .9 {
+			maxedOutStr += fmt.Sprintf(" %d/%d B disk used for %s", fsUsage.UsedBytes, fsUsage.TotalBytes, fsUsage.GetSource())
+		}
+	}
+	// Only firecracker containers set max memory limits on the VMs
```

Comment:
can we do the disk thing only for firecracker as well? want to make sure we have more time to test this before applying it to all runners